### PR TITLE
feat: adds ability for optional rabbitMQ usage for federation

### PR DESCRIPTION
**Pull request by devanbenz**
- Instance admins should be able to decide if they want to use RabbitMQ + federation. These changes will make it so the core API can effectively "turn off" the Queue mechanism if the required environment variables are missing. I have tested this locally without the env vars and it appears to be working.

### CODE REVIEW

**Comment by devanbenz** [comment]
When we start incorporating the Producer and Consumer class in to other methods we will need to ensure that its set as an optional generic during dependency injection, ie: 
```
public SomeService(Optional<Producer> producer) {
    this.producer = producer;
}

public void someMethod() {
    producer.ifPresent(prod -> prod.sendMessage("exchange", "routingKey", "message"));
}
```

**Comment by Pdzly** [comment]
We can just simply do Service Injection i think, with like

`final Optional<Producer> producer`

Then it will be automatically wired/injected by java spring boot.

**Comment by devanbenz** [comment]
> We can just simply do Service Injection i think, with like
> 
> 
> 
> `final Optional<Producer> producer`
> 
> 
> 
> Then it will be automatically wired/injected by java spring boot.

Yep that's what I'm thinking

File: src/main/java/com/sublinks/sublinksapi/queue/QueueConfig.java [hunk 1]
```diff
@@ -9,28 +9,31 @@
...
 import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 public class QueueConfig {
-  @Value("${BACKEND_QUEUE_NAME}")
+  @Value("${BACKEND_QUEUE_NAME:}")
```
Comment (Pdzly) [inline src/main/java/com/sublinks/sublinksapi/queue/QueueConfig.java:18]: Are those then ENV variables? Why not application properties?

Comment (devanbenz) [reply src/main/java/com/sublinks/sublinksapi/queue/QueueConfig.java:18]: I can change 👍

Comment (Pdzly) [reply src/main/java/com/sublinks/sublinksapi/queue/QueueConfig.java:18]: We would preferably be all in the application properties. Just to sort them all in one place and not spread small crumbs of env there and there.

**Review by Pdzly — APPROVED** [review]

**Pull request merged**